src/Draco.Compiler/Internal/Syntax/Lexer.cs

```diff
@@ -244,16 +244,63 @@ Unit TakeWithText(TokenType tokenType, int length)
         // first character
         if (char.IsDigit(ch))
         {
+            // Check for what kind of integer do we have
+            var c = this.Peek(1);
+            // Hexadecimal or binary integer
+            if (c == 'x' || c == 'b')
+            {
+                this.Advance(2);
+                var view = this.ParseIntLiteral(c == 'x' ? 16 : 2, out var value);
+                this.tokenBuilder
+                    .SetType(TokenType.LiteralInteger)
+                    .SetText($"0{c}{view}")
+                    .SetValue(value);
+                return default;
+            }
             var offset = 1;
+            var isScientificNotation = false;
             while (char.IsDigit(this.Peek(offset))) ++offset;
+            if (char.ToLower(this.Peek(offset)) == 'e')
+            {
+                isScientificNotation = true;
+                if (this.Peek(offset + 1) == '+' || this.Peek(offset + 1) == '-') offset++;
+                if (!char.IsDigit(this.Peek(offset + 1)))
+                {
+                    this.AddError(
+                        template: SyntaxErrors.UnexpectedFloatingPointLiteralEnd,
+                        offset: offset + 1,
+                        width: 1);
+                    this.tokenBuilder
+                        .SetType(TokenType.LiteralFloat)
+                        .SetText(this.Advance(offset + 1).Span.ToString());
+                    return default;
+                }
+            }
 
             // Floating point number
-            if (this.Peek(offset) == '.' && char.IsDigit(this.Peek(offset + 1)))
+            if ((this.Peek(offset) == '.' && char.IsDigit(this.Peek(offset + 1))) || isScientificNotation)
```

Comment:
That `scientificNotation` is really hard to notice, I suggest moving it to the front